syntax = "proto3";

package transport_catalogue_serialize;

message Edge {
  uint32 from = 1;
  uint32 to = 2;
  double weight = 3;
  uint32 bus_id = 4;
  uint32 stop_id = 6;
  uint32 span_count = 7;
}

message IncidenceList {
  repeated uint32 edge_id = 1;
}

message Graph {
  repeated Edge edges = 1;
  repeated IncidenceList incidence_lists = 2;
}